### PR TITLE
[libc] newheadergen: created sys folder for yaml files, make appropriate updates to CMake file for sys folder

### PR DESCRIPTION
**Pull request by RoseZhang03**
Moved sys yaml files into the sys folder.
After CMake patch lands, will make appropriate changes to account for
yaml, header, and .h.def files that are located within the sys folder in a separate patch.

### CODE REVIEW

File: libc/newhdrgen/yaml_to_classes.py [hunk 1]
```diff
@@ -232,6 +232,15 @@ def main(yaml_file, h_def_file, output_dir, add_function=None):
     output_file_name = Path(h_def_file).stem
     output_file_path = Path(output_dir) / output_file_name
 
+    # with open(yaml_file, "r") as f:
```
Comment (michaelrj-google) [inline libc/newhdrgen/yaml_to_classes.py:235]: did you mean to leave this in?

Comment (RoseZhang03) [reply libc/newhdrgen/yaml_to_classes.py:235]: Just as a placeholder for now while I wait for the CMake to come out. I'm not sure if I can manually add sys to the file path here or if I will have to make an alternate path in the CMake file.

Comment (michaelrj-google) [reply libc/newhdrgen/yaml_to_classes.py:235]: in that case I'd say it's probably best to leave this out of this patch. Commented out code is often confusing when reading a file

**Review by michaelrj-google — APPROVED** [review]
LGTM, feel free to land after addressing comments

**Pull request merged**